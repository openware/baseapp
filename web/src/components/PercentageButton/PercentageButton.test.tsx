import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { spy } from 'sinon';
import { PercentageButton, PercentageButtonProps } from '.';

const onClickSpy = spy();
const defaultProps: PercentageButtonProps = {
    value: 100,
    className: 'cr-button-percentage-100',
    onClick: onClickSpy,
};

const renderComponent = (props: Partial<PercentageButtonProps> = {}) =>
    render(<PercentageButton {...{ ...defaultProps, ...props }} />);

describe('Close Button', () => {
    it('should render', () => {
        expect(renderComponent({ onClick: onClickSpy }).container).toMatchSnapshot();
    });

    it('should call onClick callback', () => {
        renderComponent({ onClick: onClickSpy });
        fireEvent.click(screen.getByRole('button'));
        expect(onClickSpy.calledOnce).toBeTruthy();
    });
});
