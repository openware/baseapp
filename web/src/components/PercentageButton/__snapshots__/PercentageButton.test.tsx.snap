// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Close Button should render 1`] = `
<div>
  <button
    class="cr-button-percentage-100"
  >
    10000%
  </button>
</div>
`;
